kernel/src/grant.rs

```diff
@@ -42,28 +39,48 @@ impl<'a, T: 'a + ?Sized> DerefMut for Borrowed<'a, T> {
     }
 }
 
+/// A grant applied to a particular process.
+///
+/// This is created from a `Grant` to setup memory for that grant in a specific
+/// process's grant region. An `AppliedGrant` guarantees that the memory for the
+/// grant has been allocated in the process's memory.
 pub struct AppliedGrant<'a, T: 'a> {
+    /// The process the grant is applied to.
     process: &'a dyn ProcessType,
+
+    /// The identifier of the Grant this is applied for.
     grant_num: usize,
-    grant: &'a mut T,
+
+    /// Used to keep the Rust type of the grant.
     _phantom: PhantomData<T>,
 }
 
 impl<'a, T: Default> AppliedGrant<'a, T> {
-    fn get_or_allocate(grant: &Grant<T>, process: &'a dyn ProcessType) -> Result<Self, Error> {
-        // Here is an example of how the grants are laid out in a
-        // process's memory:
+    /// Create an `AppliedGrant` for the given Grant and apply it to the given
```

Comment:
What does it mean to "apply" a grant? Can you create a grant and not "apply" it?

Reply:
Apply currently means actually allocate memory from a specific process to "apply" that grant to that process.